Classes/TextUtils.cs

```diff
@@ -84,14 +90,52 @@ public static string StripDateFromName(string name, DateTime stripDate)
                     // Check if date is within the +/- 'similardateday' range of 'date'
                     if (dateFound >= stripDate.AddDays(-similarDateDayRange) && dateFound <= stripDate.AddDays(similarDateDayRange))
                     {
-                       name = matchStripDate.Replace(name, string.Empty).ToString().Trim();
+                        name = matchStripDate.Replace(name, string.Empty).ToString().Trim();
                     }
                 }
             }
 
             return name;
         }
 
+        /// <summary>
+        /// Build ‘SmartName’ from the ProgrammeName and EpisodeName values
+        /// Use regex to remove a number of different any duplication of the Programme Name.
+        /// Includes removal of date from episodeName
+        /// </summary>
+        /// <param name="programmeName">Programme Name to be checked.</param>
+        /// <param name="episodeName">Episode Name to be used to form SmartName.</param>
+        /// <param name="stripDate">Date value to be striped from the Episode Name.</param>
+        /// <returns>Episode name with date content removed</returns>
+        public static string BuildEpisodeSmartName(string programmeName, string episodeName, DateTime stripDate)
+        {
+            // check if programme name exist in episode name then remove it and any delimiting and white space that follows
+            episodeName = StripDateFromName(episodeName, stripDate).ToString().Trim();
+
+            Regex matchStripProgNameFromEpisode = new Regex(@"^(?:" + programmeName + ")" + MatchEpNameDelim + MatchEpNameWS2, RegexOptions.IgnoreCase);
```

Comment:
ok need to research this to make i have right understanding.

Reply:
I can see that escaping the variable will ensure that any special characters `. $ ^ { [ ( | ) * + ? |` within the variable are treated literally as characters rather than their unescaped function.

Is it as simple as using `Regex.Escape(programmeName)` in place of `programmeName`
Using this I can see that `Programme 6` is transformed into `Programme\\ 6` (which I'm thinking is one too many backslashes... but it seems to not throw errors and current tests still pass)

Is this what you had in mind?

Reply:
Yes, that is the function to use for escaping - I suspect that the double backslash is just how Visual Studio is displaying it to you but I may be wrong.  If the tests pass then sounds like it is working :+1: .  As belt and braces you could add a test in which the programme name being removed contains some regex metacharacters.

Reply:
Have added a couple of tests using different characters... the tests pass so hopefully all good.
Will update the PR so you can have a look over and see that the changes make sense.
Will do a few more days of testing with my podcast subscriptions to see that nothing unexpected pops up.